types/parse/index.d.ts

```diff
@@ -460,7 +481,12 @@ declare global {
             addMetadata(key: string, value: any): void;
             setTags(tags: Record<string, any>): void;
             addTag(key: string, value: any): void;
-            readonly _url: string;
+            /** url might be undefined if file is not saved */
+            readonly _url?: string;
+            readonly _data?: string;
+            readonly _source: FileSource;
+            readonly _metadata: Record<string, any>;
+            readonly _tags: Record<string, any>;
```

Comment:
It seems like these are intended to be private/internal, since there are corresponding getter functions to access them.